examples/react-whiteboard/src/App.js

```diff
@@ -104,31 +108,19 @@ function Canvas({ shapes }) {
         onPointerMove={onCanvasPointerMove}
         onPointerUp={onCanvasPointerUp}
       >
-        {Array.from(shapes, ([shapeId, shape]) => {
-          let selectionColor =
-            selectedShape === shapeId
-              ? "blue"
-              : others
-                  .toArray()
-                  .some((user) => user.presence?.selectedShape === shapeId)
-              ? "green"
-              : undefined;
-
+        {shapeIds.map((shapeId) => {
           return (
             <Rectangle
               key={shapeId}
               id={shapeId}
               onShapePointerDown={onShapePointerDown}
-              shape={shape}
-              selectionColor={selectionColor}
-              transition={selectedShape !== shapeId}
             />
           );
         })}
       </div>
       <div className="toolbar">
-        <button onClick={insertRectangle}>Rectangle</button>
-        <button onClick={deleteRectangle} disabled={selectedShape == null}>
+        <button onClick={() => insertRectangle()}>Rectangle</button>
```

Comment:
Any reason not to keep it at this? Not that it's super important, but `() => ...()` would undo the benefit of the memorized callback. In the case of the `selectedShape` on the line below, you don't have a choice though. Done for consistency/symmetry with that line?

```suggestion
        <button onClick={insertRectangle}>Rectangle</button>
```

Reply:
Not for symmetry, it's just me being overly cautious! As standard practice I don't pass parameters to functions that don't use any, in case the function changes later. This passes on a click event. If `insertRectangle` were changed to take an optional parameter, this would cause a bug.

You're totally right though, I didn't think about performance! What do you think?

Reply:
I see! I didnt even think of it that way because I’d rely so heavily on TypeScript to catch bugs like that. It doesnt have enough of a performance gain to be important here, so I’m perfectly okay with your version. Was mostly commenting out of curiosity!